ckan/lib/search/query.py

```diff
@@ -89,6 +89,27 @@ def convert_legacy_parameters_to_solr(
     return solr_params
 
 
+def _get_local_query_parser(q: str) -> str:
+    """
+    Given a Solr `q` parameter, extract any custom query parsers used in the
+    local parameters, .e.g. q={!child ...}...
+    """
+    qp_type = ""
+    q = q.strip()
+    if not q.startswith("{!"):
+        return qp_type
+
+    parts = q.lstrip("{!").rstrip("}").split(" ")
```

Comment:
local parameters can be combined with parameter values by putting the value after the `}` https://solr.apache.org/guide/solr/latest/query-guide/local-params.html#specifying-the-parameter-value-with-the-v-key so we should split on `}` here instead, right?

Reply:
Also we can't split on " " because spaces might appear in a quoted parameter something like `q={!v='lies type= here' type=dismax}` I don't know if we can avoid a full parser :-(